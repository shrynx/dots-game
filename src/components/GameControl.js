// @flow
import React from 'react';
import styled from 'styled-components';
import playButton from '../assets/play.svg';
import pauseButton from '../assets/pause.svg';
import restartButton from '../assets/restart.svg';

type Props = {
  gameState: 'Paused' | 'Playing' | 'Game Over',
  toggleGameState: () => void,
  restartGame: () => void,
};

const ControlButton = styled.button``;

const ButtonIcon = styled.img`
  height: 70px;
  width: 70px;
`;

const GameControl = ({ gameState, toggleGameState, restartGame }: Props) => {
  switch (gameState) {
    case 'Paused':
      return (
        <ControlButton onClick={toggleGameState}>
          <ButtonIcon src={playButton} alt="play" />
        </ControlButton>
      );

    case 'Playing':
      return (
        <ControlButton onClick={toggleGameState}>
          <ButtonIcon src={pauseButton} alt="pause" />
        </ControlButton>
      );

    default:
      return (
        <ControlButton onClick={restartGame}>
          <ButtonIcon src={restartButton} alt="restart" />
        </ControlButton>
      );
  }
};

export default GameControl;
