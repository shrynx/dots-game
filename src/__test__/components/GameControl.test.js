import React from 'react';
import { render } from 'react-testing-library';
import GameControl from '../../components/GameControl';

it('GameControl component Paused State should render correctly', () => {
  const { container } = render(<GameControl gameState={'Paused'} />);
  expect(container).toMatchSnapshot();
});

it('GameControl component Playing State should render correctly', () => {
  const { container } = render(<GameControl gameState={'Playing'} />);
  expect(container).toMatchSnapshot();
});

it('GameControl component Game Over State should render correctly', () => {
  const { container } = render(<GameControl gameState={'Game Over'} />);
  expect(container).toMatchSnapshot();
});
