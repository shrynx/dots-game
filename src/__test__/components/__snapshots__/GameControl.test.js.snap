// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameControl component Game Over State should render correctly 1`] = `
<div>
  <button
    class="sc-bdVaJa bDWFJH"
  >
    <img
      class="sc-bwzfXH kHVPjs"
      src="restart.svg"
      alt="restart"
    />
  </button>
</div>
`;

exports[`GameControl component Paused State should render correctly 1`] = `
<div>
  <button
    class="sc-bdVaJa bDWFJH"
  >
    <img
      class="sc-bwzfXH kHVPjs"
      src="play.svg"
      alt="play"
    />
  </button>
</div>
`;

exports[`GameControl component Playing State should render correctly 1`] = `
<div>
  <button
    class="sc-bdVaJa bDWFJH"
  >
    <img
      class="sc-bwzfXH kHVPjs"
      src="pause.svg"
      alt="pause"
    />
  </button>
</div>
`;
